task-sdk/pyproject.toml

```diff
@@ -65,6 +65,13 @@ dependencies = [
     "requests>=2.31.0,<3",
     "types-requests>=2.31.0",
     "tenacity>=8.3.0",
+    # Beginning of automatically included dependencies from shared airflow packages
```

Comment:
I'm guessing you added something in pre-commit to manage this, but I don't see it in this PR?

Reply:
Or was this meant as an examplar?

Reply:
> I'm guessing you added something in pre-commit to manage this, but I don't see it in this PR?

Yeah. That's the one part that is **not** implemented yet - I wrote about it at the beginning of the description :). ... It will require a bit of work to make it right (you need to detect who uses what and then sync the deps - including likely both "required", "dev" and "optional" deps (that would be best for cases like for example "serialization" having optional "pandas" requirement for example if we decide to leave the pandas code in serialization rather than implement some plugin mechanism for serialization, so I did not want to implement it **yet** before we agree that it's the way to go.